File: app/src/main/java/com/jrk/mood4food/model/local/LocalStorage.java
package com.jrk.mood4food.model.local;

import android.content.Context;
import android.content.SharedPreferences;

import java.util.ArrayList;
import java.util.List;

public class LocalStorage {

    public static List<LocalEntity> getAll(Context context, Class entityClass/*, filter */){

        List<LocalEntity> entities = new ArrayList<>();
        try {
            LocalEntity entity = (LocalEntity) entityClass.newInstance();
            if(entity.hasEntitySet()) {
                SharedPreferences entitySetFile = context.getSharedPreferences(entityClass.getName(), 0);
                for (String key : entitySetFile.getAll().keySet()) {
                    try {
                        entity.loadFromLocalStorage(key, entity);
                        entities.add(entity);
                    } catch (IllegalAccessException e) {
                        e.printStackTrace();
                    }
                }
            }else{
                entity.loadFromLocalStorage(entityClass.getName(), entity);
                entities.add(entity);
            }
        } catch (IllegalAccessException | InstantiationException e) {
            e.printStackTrace();
        }
        return entities;

    }

}

File: app/src/main/java/com/jrk/mood4food/model/local/LocalStorage.kt
package com.jrk.mood4food.model.local

import android.content.Context
import java.util.*

object LocalStorage {
    fun getAll(context: Context, entityClass: Class<*> /*, filter */): List<LocalEntity> {
        val entities: MutableList<LocalEntity> = ArrayList()
        try {
            val entity = entityClass.newInstance() as LocalEntity
            if (entity.hasEntitySet()) {
                val entitySetFile = context.getSharedPreferences(entityClass.name, 0)
                for (key in entitySetFile.all.keys) {
                    entity.loadFromLocalStorage(key, entity)
                    entities.add(entity)
                }
            } else {
                entity.loadFromLocalStorage(entityClass.name, entity)
                entities.add(entity)
            }
        } catch (e: IllegalAccessException) {
            e.printStackTrace()
        } catch (e: InstantiationException) {
            e.printStackTrace()
        }
        return entities
    }
}